freeipa/src/main/java/com/sequenceiq/freeipa/controller/UserV1Controller.java

```diff
@@ -34,39 +30,59 @@
     @Inject
     private PasswordService passwordService;
 
-    @Inject
-    private CrnService crnService;
-
     @Inject
     private ThreadBasedUserCrnProvider threadBaseUserCrnProvider;
 
     @Override
     public SynchronizeUserResponse synchronizeUser(SynchronizeUserRequest request) {
         String userCrn = checkUserCrn();
         LOGGER.debug("synchronizeUser() requested for user {}", userCrn);
-        String accountId = crnService.getCurrentAccountId();
+        String accountId = threadBaseUserCrnProvider.getAccountId();
 
         return userService.synchronizeUser(accountId, userCrn, userCrn);
     }
 
     @Override
-    public SynchronizeAllUsersResponse synchronizeAllUsers(SynchronizeAllUsersRequest request) {
+    public SynchronizeUserResponse getSynchronizeUserStatus(String operationId) {
+        if (operationId == null) {
+            throw new IllegalArgumentException("operationId must be provided");
+        }
+        try {
+            LOGGER.debug("getSynchronizeUserStatus() requested for operation {}", operationId);
+            return userService.getSynchronizeUsersStatus(operationId);
+        } catch (Exception e) {
+            LOGGER.error("getSynchronizeUserStatus caught exception. rethrowing", e);
```

Comment:
this is not really necessary this way as we have exceptionmappers for controllers which log the errors